src/Microsoft.DotNet.Tools.Compiler.Fsc/Program.cs

```diff
@@ -0,0 +1,145 @@
+
+                    syntax.DefineParameterList("source-files", ref sources, "Compilation sources");
+                    if (tempOutDir == null)
+                    {
+                        syntax.ReportError("Option '--temp-output' is required");
+                    }
+                });
+            }
+            catch (ArgumentSyntaxException)
+            {
+                return ExitFailed;
+            }
+
+            var translated = TranslateCommonOptions(commonOptions);
+
+            var allArgs = new List<string>(translated);
+            allArgs.AddRange(GetDefaultOptions());
+
+            // Generate assembly info
+            var assemblyInfo = Path.Combine(tempOutDir, $"dotnet-compile.assemblyinfo.fs");
+            File.WriteAllText(assemblyInfo, AssemblyInfoFileGenerator.GenerateFSharp(assemblyInfoOptions));
+            allArgs.Add($"\"{assemblyInfo}\"");
+
+            if (outputName != null)
+            {
+                allArgs.Add($"--out:\"{outputName}\"");
+            }
+
+            allArgs.AddRange(references.Select(r => $"-r:\"{r}\""));
+            allArgs.AddRange(resources.Select(resource => $"--resource:{resource}"));
+            allArgs.AddRange(sources.Select(s => $"\"{s}\""));
+
+            // Execute FSC!
+            var result = RunFsc(string.Join(" ", allArgs))
+                .ForwardStdErr()
+                .ForwardStdOut()
+                .Execute();
+
+            return result.ExitCode;
+        }
+
+        // TODO: Review if this is the place for default options
+        private static IEnumerable<string> GetDefaultOptions()
+        {
+            var args = new List<string>()
+            {
+                "--noframework",
+                "--nologo"
+            };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                args.Add("--debug:full");
+
+            return args;
+        }
+
+        private static IEnumerable<string> TranslateCommonOptions(CommonCompilerOptions options)
+        {
+            List<string> commonArgs = new List<string>();
+
+            if (options.Defines != null)
+            {
+                commonArgs.AddRange(options.Defines.Select(def => $"-d:{def}"));
+            }
+
+            if (options.Platform != null)
+            {
+                commonArgs.Add($"--platform:{options.Platform}");
+            }
+
+            if (options.WarningsAsErrors == true)
+            {
+                commonArgs.Add("--warnaserror");
+            }
+
+            if (options.Optimize == true)
+            {
+                commonArgs.Add("--optimize");
+            }
+
+            if (options.EmitEntryPoint != true)
+            {
+                commonArgs.Add("--target:library");
+            }
+
+            return commonArgs;
+        }
+
+        private static Command RunFsc(string fscArgs)
+        {
+            // Locate CoreRun
+            return Command.Create("fsc", fscArgs);
```

Comment:
I use fsc in PATH for now, no released version of fcs
At least i can use dotnet cli to compile existing f# project and try project.json
Is not the final coreclr ready implementation but is the starting point